units/pod_initializing.go

```diff
@@ -87,11 +86,18 @@ func (j *podInitJob) Run(ctx context.Context) {
 
 	switch j.pod.Status {
 	case pod.StatusInitializing:
-		if _, err := j.ecsClient.RunTask(ctx, &ecs.RunTaskInput{
-			Cluster:        utility.ToStringPtr(j.pod.Resources.Cluster),
-			ReferenceId:    utility.ToStringPtr(j.PodID),
-			TaskDefinition: utility.ToStringPtr(j.TaskID),
-		}); err != nil {
+		execOpts := cocoa.NewECSPodExecutionOptions().
+			SetCluster(j.pod.Resources.Cluster).
+			SetExecutionRole(j.env.Settings().Providers.AWS.Pod.ECS.ExecutionRole)
+
+		opts := cocoa.NewECSPodCreationOptions().
+			SetContainerDefinitions([]cocoa.ECSContainerDefinition{
+				cloud.ExportPodContainerDef(j.pod.TaskContainerCreationOpts, j.pod.Resources.SecretIDs),
```

Comment:
There seems to be a misunderstanding here about what each field in the pod document means. The `Resources.SecretIDs` don't exist yet - you have to set them after calling `CreatePod` and they have to be the ARNs of the secrets in Secrets Manager, which we can only know after `CreatePod` is called (using `ECSPod.Info(ctx)`. The secrets passed to `EnvSecrets` are assumed to be owned by the pod.

Reply:
Ah okay that makes sense, thanks for the clarification!